### DISCUSSION TITLE
blog/2021/11/08/beta-regression-guide/index

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#63] [Blog comments]
# blog/2021/11/08/beta-regression-guide/index

Everything you ever wanted to know about beta regression! Use R and brms to correctly model proportion data, and learn all about the beta distribution along the way.

https://www.andrewheiss.com/blog/2021/11/08/beta-regression-guide/index.html

**Comment by igorkf** [comment]
Hey, this is the best beta distribution post ever!

How about this approach for tricking with the (0, 1) bound?
Have you ever used this?
```r
Smithson and Verkuilen (2006)
smithson_verkuilen <- function(y) {
  n <- length(y)
  return((y * (n - 1) + 0.5) / n)
}
```
For my data it "plays" pretty well, but I'm not sure whether should I use this approach:
```r
> summary(data$resp)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.03774 0.17241 0.21562 0.34615 0.94118 
> summary(data$resp)
     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
0.0003636 0.0380720 0.1726520 0.2158268 0.3462657 0.9408556
```

**Comment by KeterAlfred** [comment]
Nice presentation. How would you interpret the Group level effects in the random effects beta regression/zero-inflated beta regression model for both the mean and the precision parameters? How would you compute the intra-class correlation coefficient (ICC)?

**Comment by eggsbrown** [comment]
Love that you covered fractional logistic regression. I use this all the time. Usually I account for "0-1 inflation" using a fractional logistic regression + observation level random effects. This approach is described here -- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4517959/

**Comment by sidmehta1975** [comment]
Thanks for a very clear tutorial professor! Also, very grateful for all the hard work you put into making your code available to the public. Can’t thank you enough.

**Comment by RaymondLTremblay** [comment]
Hello Andrew

Thank you VERY much for going through this step by step.   It is really appreciated.   Your Bayesian approach is super cool, mostly that you include a zero-inflated approach.  I work with evolutionary processes in orchids and most plant have 0% fruit set, so it will be very useful.   I'll clearly will use your website in the future. 

On another note would you have a method of doing model selection and evaluating all the combination of variables, something like the dredge function in MuMIn package?    That is my present challenge.   Much appreciated Raymond

**Comment by tylerz-145** [comment]
I am applying this code to my own data. I am having an issue with the marginaleffects() in the code. It keeps giving me the error message:

Error in typical(Holding = c(13, 26), model = list(coefficients = list( : 
  could not find function "typical"

Do you happen to know why this may be? Any help is much appreciated!

**Reply by andrewheiss** [reply]
This is because {marginaleffects} has matured a lot since November 2021. It no longer uses `typical()`, but `datagrid()` instead. I've update the post to reflect that.

**Comment by tylerz-145** [comment]
May be a dumb question, but for the Beta regression or Bayesian Beta regression, is there a way to simply output the 95% CI around the mu estimates for each group? I know we have the the mu and phi values, but not sure how to code just simply getting the 95% CI for each mu.

**Reply by tylerz-145** [reply]
* credible intervals for the Bayesian model

**Reply by andrewheiss** [reply]
You can use something like `median_hdi()` or `mean_qi()` or other related functions from {tidybayes} (see <http://mjskay.github.io/tidybayes/articles/tidybayes.html#point-summaries-and-intervals-with-the-point_interval-functions-medianmeanmode_qihdi>):

```r
model_beta_bayes %>% 
  gather_draws(`b_.*`, regex = TRUE) %>% 
  median_hdi()
```

**Reply by andrewheiss** [reply]
And that works with other dplyr functions, so you can group by specific columns:

```r
model_beta_bayes %>% 
  gather_draws(`b_.*`, regex = TRUE) %>% 
  group_by(whatever) %>%
  median_hdi()
```

**Reply by Miki-E** [reply]
This is literally one of the best descriptions of beta and inflated beta models I have ever seen!

Well done.

Is it at all possible to add something of the same quality describing how we test if the zero inflated beta is a good fit, and possibly some bayesian comparison (BIC or DIC) for chosing which zero inflated beta model covariates to keep in the model?

**Comment by Crispy13** [comment]
Thank you for posting!

It helped me a lot in my work.

**Comment by Evattimo** [comment]
Hi. This explanation is amazing! However, I understood that the explanation regarding the 0-inflated beta regression only uses the categorical variable (quota) as an example. My variable is continuous, which is addressed by the regular beta regression in this tutorial. I did not see any example for a zero-inflated beta regression with a continuous covariate. It would be very helpful if we could discuss that. Thank you.
